### PR TITLE
[Driver][SYCL] Generalize huge device code driver option name

### PR DESCRIPTION
**Pull request by tcreech-intel**
Deprecate -fsycl-link-huge-device-code in favor of a new option,
-flink-huge-device-code. The new option is identical in functionality
but allowed with -fopenmp-targets.

This also includes a minor improvement to deprecated handling in order
to allow aliasing driver options to be deprecated independently.

### CODE REVIEW

File: clang/test/Driver/sycl-huge-device-code-linker-script.cpp [hunk 1]
```diff
@@ -3,10 +3,15 @@
...
+// RUN:       FileCheck --check-prefix=CHECK-LINKER-SCRIPT %s
 // CHECK-LINKER-SCRIPT: "-T" "{{.*}}.ld"
 
 // Also check that a user-provided linker script may be used:
 // RUN: %clangxx -### -fsycl -fsycl-link-huge-device-code %s \
 // RUN:       -T custom-user-script.ld 2>&1 | \
 // RUN:       FileCheck --check-prefixes=CHECK-USER-SCRIPT %s
+// RUN: %clangxx -### -fopenmp -fopenmp-targets=x86_64 %s \
+// RUN:       -T custom-user-script.ld 2>&1 | \
+// RUN:       FileCheck --check-prefixes=CHECK-USER-SCRIPT %s
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-huge-device-code-linker-script.cpp:16]: Test name is specific to SYCL behaviors - we could either create a new test for OpenMP behaviors or rename this test to something more fitting.

Comment (tcreech-intel) [reply clang/test/Driver/sycl-huge-device-code-linker-script.cpp:18]: Thanks. I decided to rename this test after the new option, with a comment describing that the old option name which is also being tested is deprecated.

**Review by mdtoguchi — APPROVED** [review]
LGTM

**Comment by tcreech-intel** [comment]
I don't think that [the failing test](https://github.com/intel/llvm/actions/runs/5337850111/jobs/9674790290?pr=9888#step:8:486) is related to the change made in this PR.

**Comment by tcreech-intel** [comment]
@intel/dpcpp-doc-reviewers, could you please take a look?

**Review by bader — APPROVED** [review]

**Comment by steffenlarsen** [comment]
HIP Failed Tests (1):
  SYCL :: Basic/event_profiling_info.cpp
This was recently enabled and I think the order issue was unexpected. Tag @mmoadeli .

Merging this.

**Comment by mmoadeli** [comment]
> HIP Failed Tests (1): SYCL :: Basic/event_profiling_info.cpp This was recently enabled and I think the order issue was unexpected. Tag @mmoadeli .
> 
> Merging this.

That's a bit odd.

**Pull request merged**